Source/missiles.cpp

```diff
@@ -674,14 +674,14 @@ BOOL MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, int t, BOOLE
 		if (resist) {
 			PlayEffect(m, 1);
 		} else if (monster[m]._mmode == MM_STONE) {
-			if (m > 3)
+			if (m >= MAX_PLRS)

```

Comment:
```suggestion
			if (m > MAX_PLRS - 1)
```
Same for the next one